modules/social_features/social_embed/src/Plugin/Filter/SocialEmbedConvertUrlToEmbedFilter.php

```diff
@@ -0,0 +1,241 @@
+      $valid_url_ending_characters = '[\p{L}\p{M}\p{N}:_+~#=/]|(?:' . $valid_url_balanced_parens . ')';
+
+      $valid_url_query_chars = '[a-zA-Z0-9!?\*\'@\(\);:&=\+\$\/%#\[\]\-_\.,~|]';
+      $valid_url_query_ending_chars = '[a-zA-Z0-9_&=#\/]';
+
+      // Full path  and allow @ in a url, but only in the middle. Catch things
+      // like http://example.com/@user/
+      $valid_url_path = '(?:(?:' . $valid_url_path_characters . '*(?:' . $valid_url_balanced_parens . $valid_url_path_characters . '*)*' . $valid_url_ending_characters . ')|(?:@' . $valid_url_path_characters . '+\/))';
+
+      // Prepare domain name pattern.
+      // The ICANN seems to be on track towards accepting more diverse top level
+      // domains, so this pattern has been "future-proofed" to allow for TLDs
+      // of length 2-64.
+      $domain = '(?:[\p{L}\p{M}\p{N}._+-]+\.)?[\p{L}\p{M}]{2,64}\b';
+      $ip = '(?:[0-9]{1,3}\.){3}[0-9]{1,3}';
+      $auth = '[\p{L}\p{M}\p{N}:%_+*~#?&=.,/;-]+@';
+      $trail = '(' . $valid_url_path . '*)?(\\?' . $valid_url_query_chars . '*' . $valid_url_query_ending_chars . ')?';
+
+      // Match absolute URLs.
+      $url_pattern = "(?:$auth)?(?:$domain|$ip)/?(?:$trail)?";
+      $pattern = "`$url_prefix((?:$protocols)(?:$url_pattern))`u";
+      $tasks['replaceFullLinks'] = $pattern;
+
+      // HTML comments need to be handled separately, as they may contain HTML
+      // markup, especially a '>'. Therefore, remove all comment contents
+      // and add them back later.
+      _filter_url_escape_comments('', TRUE);
+      $text = preg_replace_callback('`<!--(.*?)-->`s', '_filter_url_escape_comments', $text);
+
+      // Split at all tags; ensures that no tags or attributes are processed.
+      $chunks = preg_split('/(<.+?>)/is', $text, -1, PREG_SPLIT_DELIM_CAPTURE);
+      // PHP ensures that the array consists of alternating delimiters and
+      // literals, and begins and ends with a literal (inserting NULL as
+      // required). Therefore, the first chunk is always text:
+      $chunk_type = 'text';
+      // If a tag of $ignore_tags is found, it is stored in $open_tag and only
+      // removed when the closing tag is found. Until the closing tag is found,
+      // no replacements are made.
+      $open_tag = '';
+
+      for ($i = 0; $i < count($chunks); $i++) {
+        if ($chunk_type == 'text') {
+          // Only process this text if there are no unclosed $ignore_tags.
+          if ($open_tag == '') {
+            // If there is a match, inject a link into this chunk via the
+            // callback function contained in $task.
+            $chunks[$i] = preg_replace_callback(
+              $pattern,
+              function ($match) {
+                try {
+                  $info = \Drupal::service('url_embed')->getUrlInfo(Html::decodeEntities($match[1]));
+                  if ($info) {
+                    // Replace URL with consent button.
+                    $uuid = \Drupal::service('uuid')->generate();
```

Comment:
why are we generating a new uuid? isn't there a uuid we can use maybe from the content where this is placed for example? or how will we know that the uuid matches?

if that is not possible or doesn't make sense for this use case we can drop it and just allow it based on the URL that comes in. WIth proper caching in place that should also be fine.

Reply:
@ronaldtebrake the filter plugin is not aware of the entity it is processing for. Even if we had the entity, we can have multiple embeds possible. 

We can get nodes from route parameters in the process function, but there can be multiple posts in the stream, also such methods will be costly on performance. And even if I get the entity, we can have multiple links, so we will have to assign unique identifiers to each of the divs so that we can replace the same from AjaxResponse in Controller.

The reason I chose this method is to add unique ids in each div we are creating for an embeddable link and also, we want to replace that particular div only. 

But you are right, as of now, there is no way we can match the UUIDs set on the divs in the controller, we can just validate it.

The other way is to use the URL itself as an ID.